parlai/mturk/core/mturk_data_handler.py

```diff
@@ -0,0 +1,474 @@
+        c = conn.cursor()
+        c.execute('''UPDATE pairings SET status = ?, onboarding_start = ?
+                     WHERE worker_id = ? AND assignment_id = ?;''',
+                  (AssignState.STATUS_ONBOARDING, time.time(), worker_id,
+                   assignment_id))
+        conn.commit()
+
+    def log_finish_onboard(self, worker_id, assignment_id):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('''UPDATE pairings SET status = ?, onboarding_end = ?
+                     WHERE worker_id = ? AND assignment_id = ?;''',
+                  (AssignState.STATUS_WAITING, time.time(), worker_id,
+                   assignment_id))
+        conn.commit()
+
+    def log_start_task(self, worker_id, assignment_id, conversation_id):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('''UPDATE pairings SET status = ?, task_start = ?,
+                     conversation_id = ? WHERE worker_id = ?
+                     AND assignment_id = ?;''',
+                  (AssignState.STATUS_IN_TASK, time.time(), conversation_id,
+                   worker_id, assignment_id))
+        conn.commit()
+
+    def log_award_amount(self, worker_id, assignment_id, amount, reason):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('''UPDATE pairings SET bonus_amount = ?, bonus_text = ?
+                     WHERE worker_id = ? AND assignment_id = ?;''',
+                  (amount, reason, worker_id, assignment_id))
+        conn.commit()
+
+    def log_bonus_paid(self, worker_id, assignment_id):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('''UPDATE pairings SET bonus_paid = ?
+                     WHERE worker_id = ? AND assignment_id = ?;''',
+                  (True, worker_id, assignment_id))
+        conn.commit()
+
+    def log_approve_assignment(self, worker_id, assignment_id):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('UPDATE assignments SET status = ? WHERE id = ?;',
+                  ('Approved', assignment_id))
+        c.execute('''UPDATE workers SET approved = approved + 1
+                     WHERE id = ?;''',
+                  (worker_id, ))
+        conn.commit()
+
+    def log_reject_assignment(self, worker_id, assignment_id):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('UPDATE assignments SET status = ? WHERE id = ?;',
+                  ('Rejected', assignment_id))
+        c.execute('''UPDATE workers SET rejected = rejected + 1
+                     WHERE id = ?;''',
+                  (worker_id, ))
+        conn.commit()
+
+    def get_worker_data(self, worker_id):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute("SELECT * FROM workers WHERE id = ?;", (worker_id, ))
+        results = c.fetchone()
+        if results is None:
+            return None
+        return {
```

Comment:
looking at https://docs.python.org/3/library/sqlite3.html#sqlite3.Row, it looks like you can set the `row_factory` to be `sqlite3.Row`, from which you can access the keys when you call `fetchone()`. might be helpful instead of these long dicts :)

Reply:
also, i noticed you returned the dict here rather than making this a `_format..` method

Reply:
Yeah I think I can make the changes to have the row_factory solution work, thanks for finding this!